### PR TITLE
docs: render china mirror link before hydration

### PR DESCRIPTION
**Pull request by afc163**
[[中文版模板 / Chinese template](https://github.com/ant-design/ant-design/blob/master/.github/PULL_REQUEST_TEMPLATE/pr_cn.md?plain=1)]

### 🤔 This is a ...

- [ ] New feature
- [ ] Bug fix
- [x] Site / documentation update
- [ ] Demo update
- [ ] Component style update
- [ ] TypeScript definition update
- [ ] Bundle size optimization
- [ ] Performance optimization
- [ ] Enhancement feature
- [ ] Internationalization
- [ ] Refactoring
- [ ] Code style optimization
- [ ] Test Case
- [ ] Branch merge
- [ ] Workflow
- [ ] Other (about what?)

### 💡 Background and solution

两个优化，避免只有等待 js 注水后国内镜像链接才能出来。网速慢的情况下基本等不到 js 加载完。

- [x] 服务端渲染就加上国内镜像，而不是客户端才渲染。
- [x] submenu 可以直接点击打开镜像链接。

### 📝 Changelog

| Language   | Changelog |
| ---------- | --------- |
| 🇺🇸 English |    --       |
| 🇨🇳 Chinese |   --        |

### ☑️ Self-Check before Merge

⚠️ Please check all items below before requesting a reviewing. ⚠️

- [x] Doc is updated/provided or not needed
- [x] Demo is updated/provided or not needed
- [x] TypeScript definition is updated/provided or not needed
- [x] Changelog is provided or not needed

---

### <samp>🤖 Generated by Copilot at 11e1578</samp>

Improved the dumi theme header and navigation. Updated the `Navigation` component to use the `colorText` token and link elements. Removed the unused `isClient` prop from the `Header` and `Navigation` components.

### <samp>🤖 Generated by Copilot at 11e1578</samp>

*  Remove `isClient` prop from `Header` and `Navigation` components as it is no longer needed after migrating to `@umijs/preset-dumi` package ([link](https://github.com/ant-design/ant-design/pull/44233/files?diff=unified&w=0#diff-2b1224dbeb7d20f1cb1b4590eb12b68104c605d2a67f0b0bd4fd9d8b0100a728L276), [link](https://github.com/ant-design/ant-design/pull/44233/files?diff=unified&w=0#diff-632020f2d1bdd2829b18be604bbd84f96a304722cc050a8584a90d760699b4c5L100), [link](https://github.com/ant-design/ant-design/pull/44233/files?diff=unified&w=0#diff-632020f2d1bdd2829b18be604bbd84f96a304722cc050a8584a90d760699b4c5L109))
*  Add `colorText` token to `useStyle` function and use it to style the links in the navigation menu with a transparent pseudo-element for better accessibility ([link](https://github.com/ant-design/ant-design/pull/44233/files?diff=unified&w=0#diff-632020f2d1bdd2829b18be604bbd84f96a304722cc050a8584a90d760699b4c5L33-R34), [link](https://github.com/ant-design/ant-design/pull/44233/files?diff=unified&w=0#diff-632020f2d1bdd2829b18be604bbd84f96a304722cc050a8584a90d760699b4c5R60-R70))
*  Change the `label` prop of the `mirror` and `Gitee 镜像` menu items from plain strings to link elements to make them consistent with other items and allow opening the mirror sites in new tabs or windows ([link](https://github.com/ant-design/ant-design/pull/44233/files?diff=unified&w=0#diff-632020f2d1bdd2829b18be604bbd84f96a304722cc050a8584a90d760699b4c5L228-R253), [link](https://github.com/ant-design/ant-design/pull/44233/files?diff=unified&w=0#diff-632020f2d1bdd2829b18be604bbd84f96a304722cc050a8584a90d760699b4c5L248-R269))

### CODE REVIEW

**Review by kiner-tang — APPROVED** [review]

File: .dumi/theme/slots/Header/Navigation.tsx [hunk 1]
```diff
@@ -225,15 +235,22 @@ export default ({
       key: 'docs/resources',
     },
     isZhCN &&
-    isClient &&
     window.location.host !== 'ant-design.antgroup.com' &&
     window.location.host !== 'ant-design.gitee.io'
       ? {
```
Comment (MadCcc) [inline .dumi/theme/slots/Header/Navigation.tsx:240]: 这个 SSR 渲染会报错的吧

**Comment by MadCcc** [comment]
image

**Review by MadCcc — DISMISSED** [review]
Pre-render 失败了

**Comment by afc163** [comment]
> Pre-render 失败了

ci 居然没报错：https://github.com/ant-design/ant-design/actions/runs/5864544457/job/15899878474?pr=44233

cc @PeachScript

File: .dumi/theme/slots/Header/Navigation.tsx [hunk 2]
```diff
@@ -135,6 +145,13 @@ export default ({
     activeMenuItem = 'docs/resources';
   }
 
+  const [hideChinaMirror, setHideChinaMirror] = React.useState(true);
+  React.useEffect(() => {
+    setHideChinaMirror(
+      ['ant-design.antgroup.com', 'ant-design.gitee.io'].includes(window.location.host),
+    );
+  }, []);
```
Comment (MadCcc) [inline .dumi/theme/slots/Header/Navigation.tsx:153]: 这样还是会在注水之后显示，感觉没啥意义

Comment (afc163) [reply .dumi/theme/slots/Header/Navigation.tsx:153]: 现在是反的，prerender 里默认先出来，注水之后隐藏。

Comment (MadCcc) [reply .dumi/theme/slots/Header/Navigation.tsx:153]: 那样镜像站感觉会比较怪。。我觉得不如中文页面就常显

**Comment by PeachScript** [comment]
> github action 居然没报错

刚和 @MadCcc 讨论下个版本改成抛错；之前是为了延续 Umi 3 的行为，但现在看抛错更合理

**Pull request merged**